client/transaction-pool/tests/pool.rs

```diff
@@ -49,14 +50,36 @@ fn pool() -> Pool<TestApi> {
 }
 
 fn maintained_pool() -> (BasicPool<TestApi, Block>, Arc<TestApi>, futures::executor::ThreadPool) {
-	let api = Arc::new(TestApi::with_alice_nonce(209));
-	let (pool, background_task) = BasicPool::new_test(api.clone());
+	let test_api = TestApi::with_alice_nonce(209);
+	let genesis_hash = {
+		let chain = test_api.chain().read();
+		chain
+			.block_by_number
+			.get(&0)
+			.map(|blocks| blocks[0].0.header.hash())
+			.expect("there is block 0. qed")
+	};
+	let api = Arc::new(test_api);
+	let (pool, background_task) = BasicPool::new_test(api.clone(), genesis_hash, genesis_hash);
 
 	let thread_pool = futures::executor::ThreadPool::new().unwrap();
 	thread_pool.spawn_ok(background_task);
 	(pool, api, thread_pool)
 }
 
+fn create_basic_pool(test_api: TestApi) -> BasicPool<TestApi, Block> {
```

Comment:
This is only a test, but I rather prefer to avoid duplicated code.
This new `create_basic_pool` differs from `maintained_pool` only because of the non hardcoded `TestApi`. Maybe we can let the above call this and let this return the `Future` as well?
Again, not super important but better to follow some design best practices even for the tests.

Reply:
Existing code isn't perfect, params are somehow mixed Arc vs TestApi. Done my best in [32a9522](https://github.com/paritytech/substrate/pull/12305/commits/32a95221007e6c133453706adad91a3e209901c7)